xarray/backends/zarr.py

```diff
@@ -1178,6 +1178,11 @@ def set_variables(
                 fill_value = attrs.pop("_FillValue", None)
             else:
                 fill_value = v.encoding.pop("fill_value", None)
+                if fill_value is None and v.dtype.kind == "f":
+                    # For floating point data, Xarray defaults to a fill_value
+                    # of NaN (unlike Zarr, which uses zero):
+                    # https://github.com/pydata/xarray/issues/10646
+                    fill_value = v.dtype.type(np.nan)
```

Comment:
I'm not sure whether there is any point to use a typed nan, since ultimately it just ends up as JSON. Somewhat related to https://github.com/zarr-developers/zarr-python/issues/3478

Reply:
Good point. Probably best to just use plain `np.nan` then.